diff --git a/es/_util/vnode.js b/es/_util/vnode.js
index 16db1785e001e98184d5293d35d3f3b3bc7f0163..5b5add950a4fd2db749f910dd294d92ef931a82f 100644
--- a/es/_util/vnode.js
+++ b/es/_util/vnode.js
@@ -42,3 +42,8 @@ export function deepCloneElement(vnode) {
     return cloned;
   }
 }
+
+export function triggerVNodeUpdate(instance) {
+  instance.component.effect.dirty = true;
+  instance.component.update();
+}
diff --git a/es/modal/confirm.js b/es/modal/confirm.js
index 099cf2e25e5810f90802539713a045a790af2737..0eea6c3387c802aad184191b5ea0d9a3503c39db 100644
--- a/es/modal/confirm.js
+++ b/es/modal/confirm.js
@@ -5,6 +5,7 @@ import { createVNode, render as vueRender } from 'vue';
 import ConfirmDialog from './ConfirmDialog';
 import ConfigProvider, { globalConfigForApi } from '../config-provider';
 import omit from '../_util/omit';
+import { triggerVNodeUpdate } from '../_util/vnode';
 import { getConfirmLocale } from './locale';
 import destroyFns from './destroyFns';
 const confirm = config => {
@@ -63,7 +64,7 @@ const confirm = config => {
     }
     if (confirmDialogInstance) {
       _extends(confirmDialogInstance.component.props, currentConfig);
-      confirmDialogInstance.component.update();
+      triggerVNodeUpdate(confirmDialogInstance);
     }
   }
   const Wrapper = p => {